phoenix-core/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -231,6 +231,24 @@ public static Expression pushKeyExpressionsToScan(StatementContext context, Filt
                 if (prevSortOrder == null)  {
                     prevSortOrder = sortOrder;
                 } else if (prevSortOrder != sortOrder) {
+                    //Consider the Universe of keys to be [0,7]+ on the leading column A
```

Comment:
what do you mean [0,7]+ here?

Reply:
That regular expression based that is keys are a collection of values between 0 and 7777...